### PR TITLE
net-misc/icaclient: bump to 22.12.0.12

### PR DESCRIPTION
**Pull request by henning-schild**
Here we do not just bump the version but also change the way we deal with the shared config file between this ebuild and its plugin media-plugins/hdx-realtime-media-engine.
Also we move the rdep to sys-libs/llvm-libunwind to the plugin. And if the user wants hdx support on icaclient they have to set a USE flag. The plugin will not work if icaclient is not built with support for it and the plugin itself contains the tool to enable itself so we turn the dep chain around, the plugin is even a bdep of the application.

Closes: https://bugs.gentoo.org/883775
Closes: https://bugs.gentoo.org/881637
Signed-off-by: Henning Schild <henning@hennsch.de>

### CODE REVIEW

**Comment by henning-schild** [comment]
@juippis this is open for a while now, not a simple bump but i would like to ask you to review, you have been involved in other PRs on that package

File: net-misc/icaclient/icaclient-22.12.0.12.ebuild [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	# we own it now and avoid conflict warnings with this
+	rm -f "${ROOT}${ICAROOT}/config/module.ini"
+}
+
+pkg_postinst() {
+	xdg_desktop_database_update
+	xdg_mimeinfo_database_update
+
+	if ! use hdx; then
+		if [ -x "${ROOT}${ICAROOT}"/rtme/RTMEconfig ]; then
```
Comment (thesamesam) [inline net-misc/icaclient/icaclient-22.12.0.12.ebuild:265]: Use bash tests ([[ ]])

File: net-misc/icaclient/icaclient-22.12.0.12.ebuild [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+src_unpack() {
+	default
+	mv "${WORKDIR}/${ICAARCH}/${ICAARCH}.cor" "${S}" || die
+}
+
+src_prepare() {
+	default
+	rm lib/UIDialogLibWebKit.so || die
+
+	cp nls/en/module.ini .
```
Comment (thesamesam) [inline net-misc/icaclient/icaclient-22.12.0.12.ebuild:124]: missing die on cp, rtmeconfig call, mv and mv.

**Comment by juippis** [comment]
Sorry I was away with a travel laptop that isn't suited for building these PR packages, and put my full focus on going through my unread bugzilla mail. Glad to see @thesamesam was able to process this!

**Pull request closed without merging**